pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -255,6 +255,11 @@ public void deleteSegmentFile() {
   private StreamPartitionMsgOffset _finalOffset; // Used when we want to catch up to this one
   private volatile boolean _shouldStop = false;
 
+  /** This variable will be set by the configured {@literal IngestionBasedConsumptionStatusChecker} when the segment is
+   * caughtup.
+   */
+  private volatile boolean _caughtUpWithUpstream = false;
```

Comment:
This will be true once the server is caughtup, how do you handle the scenario that a sudden traffic burst or consumer pause/resume?

Reply:
Very good questions and I think we will encounter can increase in the lag during traffic spike. However, we will be monitoring on the "trend" of the lag, where it should always be trending down or remain stable. 

The problem with pause/resume is that the segment or table data manager doesn't know if the consumer has been paused because it is simply treated as a force commit. I think we still want to have visibility when the stream is paused. But the other metric `LLC_PARTITION_CONSUMING` will not be true in this case. So, the monitoring rule should be something like:
ALERT IF (`LLC_PARTITION_CONSUMING` == 1 && `availabilityLagMs` is trending up ) for a period of 15min

Reply:
idealstate has indication that pause is on